src/main.rs

```diff
@@ -119,11 +147,27 @@ pub async fn main() {
     };
 
     let (event_read_fd, event_write_fd) = pipe2(OFlag::O_NONBLOCK).unwrap();
+    let sighup_occurred = Arc::new(AtomicBool::new(false));
+    {
+        let sighup_occurred = Arc::clone(&sighup_occurred);
+        if let Err(e) = unsafe {
```

Comment:
If you stored the closure in a variable, I think you might be able to limit the scope of `unsafe` some more?

Reply:
I'm not sure I understand?

Reply:
So if you did something like:
```
if let Err(e) = {
    let f = move || {
        // All functions called in this function must be signal safe. See signal(3).
        sighup_occurred.store(true, Ordering::Relaxed);
        unsafe { nix::unistd::write(event_write_fd, &[0]).ok() };
    };
    unsafe { signal_hook::register(signal_hook::SIGHUP, f) };
}
```

Then fewer lines can be in `unsafe`? I may be wrong.

Reply:
TBH, I'm fairly happy having the closure in `unsafe` because it's a signal handler and being clear that "here be dragons" is not a bad idea.